src/commands/deployment/get.ts

```diff
@@ -2,95 +2,101 @@ import Table from "cli-table";
 import commander from "commander";
 import Deployment from "spektate/lib/Deployment";
 import AzureDevOpsPipeline from "spektate/lib/pipeline/AzureDevOpsPipeline";
-import IPipeline from "spektate/lib/pipeline/Pipeline";
 import { Config } from "../../config";
+import { build as buildCmd, exit as exitCmd } from "../../lib/commandBuilder";
+import { isIntegerString } from "../../lib/validator";
 import { logger } from "../../logger";
-export let hldPipeline: IPipeline;
-export let clusterPipeline: IPipeline;
-export let srcPipeline: IPipeline;
+import { IConfigYaml } from "../../types";
+import decorator from "./get.decorator.json";
+
 /**
  * Output formats to display service details
  */
 export enum OUTPUT_FORMAT {
-  /**
-   * Normal format
-   */
-  NORMAL = 0,
+  NORMAL = 0, // normal format
+  WIDE = 1, // Wide table format
+  JSON = 2
+}
 
-  /**
-   * Wide table format
-   */
-  WIDE = 1,
+export interface IInitObject {
+  config: IConfigYaml;
+  clusterPipeline: AzureDevOpsPipeline;
+  hldPipeline: AzureDevOpsPipeline;
+  key: string;
+  srcPipeline: AzureDevOpsPipeline;
+}
+export interface ICommandOptions {
+  watch: boolean;
+  output: string;
+  env: string;
+  imageTag: string;
+  buildId: string;
+  commitId: string;
+  service: string;
+  deploymentId: string;
+  top: string;
+}
 
-  /**
-   * JSON format
-   */
-  JSON = 2
+export interface IValidatedOptions extends ICommandOptions {
+  nTop: number;
+  outputFormat: OUTPUT_FORMAT;
 }
 
+export const validateValues = (opts: ICommandOptions): IValidatedOptions => {
+  let top = 0; // no limits
+  if (opts.top) {
+    if (isIntegerString(opts.top)) {
+      top = parseInt(opts.top, 10);
+    } else {
+      throw new Error("value for top option has to be a positive number");
+    }
+  }
+
+  return {
+    buildId: opts.buildId,
+    commitId: opts.commitId,
+    deploymentId: opts.deploymentId,
+    env: opts.env,
+    imageTag: opts.imageTag,
+    nTop: top,
+    output: opts.output,
+    outputFormat: processOutputFormat(opts.output),
+    service: opts.service,
+    top: opts.top,
+    watch: opts.watch
+  };
+};
+
+export const execute = async (
```

Comment:
nit: jsdoc for `execute`, `validateValues`

Reply:
accept